snuba/request/request_settings.py

```diff
@@ -45,12 +105,31 @@ class HTTPRequestSettings(RequestSettings):
     """
 
     def __init__(
-        self, turbo: bool = False, consistent: bool = False, debug: bool = False
+        self,
+        turbo: bool = False,
+        consistent: bool = False,
+        debug: bool = False,
+        sampling_config: Optional[Mapping[str, Any]] = None,
     ) -> None:
         self.__turbo = turbo
         self.__consistent = consistent
         self.__debug = debug
         self.__rate_limit_params = [get_global_rate_limit_params()]
+        self.__sampling_config = (
+            SamplingConfig.build_from_dict(sampling_config)
+            if sampling_config
+            else SamplingConfig.build_default()
+        )
+
+        # TODO: Enforce we no other sampling rate except for None can be used
+        # if we are not in turbo mode
```

Comment:
I suspect there are still some use cases where the sampling is passed when not in turbo mode.